CRM/Core/BAO/CustomGroup.php

```diff
@@ -225,6 +227,22 @@ public static function retrieve(&$params, &$defaults) {
     return CRM_Core_DAO::commonRetrieve('CRM_Core_DAO_CustomGroup', $params, $defaults);
   }
 
+  /**
+   * Ensure group name does not conflict with an existing field
+   *
+   * @param CRM_Core_DAO_CustomGroup $group
+   */
+  public static function validateCustomGroupName(CRM_Core_DAO_CustomGroup $group) {
+    $extends = in_array($group->extends, CRM_Contact_BAO_ContactType::basicTypes(TRUE)) ? 'Contact' : $group->extends;
+    $extendsDAO = CRM_Core_DAO_AllCoreTables::getFullName($extends);
+    if ($extendsDAO) {
+      $fields = array_column($extendsDAO::fields(), 'name');
+      if (in_array($group->name, $fields)) {
+        $group->name .= '0';
```

Comment:
@colemanw what happens if you try to create a 2nd ActivityType Custom Group? i.e. one already has a 0 on it would that be an issue (this is probably quite the edge case but just thinking)

Reply:
The current behavior is that if you try to create a custom group with a duplicate name (or for that matter, a duplicate title) then you'll get a DB error due to the unique key constraints.

This PR doesn't alter that.